staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -300,3 +307,61 @@ func (le *LeaderElector) maybeReportTransition() {
 		go le.config.Callbacks.OnNewLeader(le.reportedLeader)
 	}
 }
+
+type LeaderChecker interface {
```

Comment:
Consider a new file in the same package for this?

Reply:
Godoc comment :)

Reply:
"LeaderChecker" is a bit vague for a name if you just look at this package it's not clear what it'd mean. How about "HealthzAdaptor"?

Reply:
How about "Watchdog"?